### PR TITLE
feat(multichain): normalize address helpers

### PR DESCRIPTION
**Pull request by ccharly**
## **Description**

> [!IMPORTANT]  
> This work is required for the BTC initiative within the extension.

New helpers to normalize address that can also be used in a non-EVM context.

From a user perspective, nothing should have changed. The changes are mainly in components that displays an address:

- - -

Screenshot 2024-06-13 at 13 29 35From a user perspective, nothing should have changed. The changes are mainly located in those areas:

- - -

Screenshot 2024-06-13 at 13 30 40

- - -

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25022?quickstart=1)

## **Related issues**

None

## **Manual testing steps**

None

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/multichain/address.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  {
+    address: '0xdeadbeef',
+  },
+  {
+    address: 'bc1ql49ydapnjafl5t2cp9zqpjwe6pdgmxy98859v2',
+  },
+];
+
+function toLowerCaseEthAddress(address: string) {
+  return `0x${stripHexPrefix(address).toLowerCase()}`;
```
Comment (gantunesr) [inline app/scripts/lib/multichain/address.test.ts:39]: Why are you removing and then adding the prefix here?
```suggestion
  return address.toLowerCase();
```

**Review by gantunesr — COMMENTED** [review]
LGTM - just left a small comment

File: app/scripts/lib/multichain/address.test.ts (hunk 1 above)
Comment (ccharly) [reply app/scripts/lib/multichain/address.test.ts:39]: ah yes, that's a leftover, cause I used to also have a `toUpperCaseEthAddress` too, that was wrongly upper-casing the `0x` 🙈 (and I was using the same pattern for both the lower/upper implementation)

So let me use this your suggestion!

**Review by gantunesr — APPROVED** [review]

**Review by montelaidev — APPROVED** [review]
Lgtm

**Pull request merged**